plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -110,6 +111,7 @@ public abstract class BaseJdbcClient
     protected final RemoteQueryModifier queryModifier;
     private final IdentifierMapping identifierMapping;
     private final boolean supportsRetries;
+    private Boolean storesUpperCaseIdentifiers;
```

Comment:
Maybe we could have some `JdbcRemoteIdentifiers.Factory`, that would store the field and here we would have a factory on a field. It would be clear that this field is needed only for that and also we can provide a comment why this caching is needed at all.

Reply:
refactored